pyFV3/stencils/delnflux.py

```diff
@@ -179,74 +113,51 @@ def fy_calculation_neg(q: FloatField, del6_u: FloatField):
     return -del6_u * (q[0, -1, 0] - q)
 
 
-def d2_highorder_stencil(
-    fx: FloatField, fy: FloatField, rarea: FloatFieldIJ, d2: FloatField
+def d2_highorder_stencil_FV3GFS(
+    fx: FloatField,
+    fy: FloatField,
+    rarea: FloatFieldIJ,
+    nord: FloatFieldK,
+    d2: FloatField,
+    current_nord: int,
 ):
-    from __externals__ import nord0, nord1, nord2, nord3
-
-    with computation(PARALLEL), interval(0, 1):
-        if __INLINED(nord0 > 0):
-            d2 = d2_highorder(fx, fy, rarea)
-    with computation(PARALLEL), interval(1, 2):
-        if __INLINED(nord1 > 0):
-            d2 = d2_highorder(fx, fy, rarea)
-    with computation(PARALLEL), interval(2, 3):
-        if __INLINED(nord2 > 0):
-            d2 = d2_highorder(fx, fy, rarea)
-    with computation(PARALLEL), interval(3, None):
-        if __INLINED(nord3 > 0):
-            d2 = d2_highorder(fx, fy, rarea)
+    with computation(PARALLEL), interval(...):
+        if nord > current_nord:
+            d2 = (fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) * rarea
 
 
-@gtscript.function
-def d2_highorder(fx: FloatField, fy: FloatField, rarea: FloatField):
-    d2 = (fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) * rarea
-    return d2
+def d2_highorder_stencil_GEOS(
+    fx: FloatField,
+    fy: FloatField,
+    rarea: FloatFieldIJ,
+    nord: FloatFieldK,
+    d2: FloatField,
+    current_nord: int,
+):
+    with computation(PARALLEL), interval(...):
+        if nord > current_nord:
+            d2 = ((fx - fx[1, 0, 0]) + (fy - fy[0, 1, 0])) * rarea
```

Comment:
At double floating precision, this has minimal to no impact. At single precision, however the order of the subtraction has measurable impacts (especially because of close-to-zero behavior of a lot of those codes).

A large amount of those diff pattern have been corrected in the GEOS fork of FV3.

Reply:
Great. We'd like to include a number of these in the GFDL FV3 also.